app/templates/gulpfile.js

```diff
@@ -255,109 +291,124 @@ gulp.task('build', ['optimize', 'images', 'fonts', 'tinymce'], function() {
 gulp.task('optimize', ['inject', 'test'], function() {
   log('Optimizing the js, css, and html');
 
-  var assets = $.useref.assets({searchPath: './'});
   // Filters are named for the gulp-useref path
-  var cssFilter = $.filter('**/*.css');
-  var jsAppFilter = $.filter('**/' + config.optimized.app);
-  var jslibFilter = $.filter('**/' + config.optimized.lib);
+  var cssFilter = $.filter('**/*.css', {restore: true});
+  var jsAppFilter = $.filter('**/' + config.optimized.app, {restore: true});
+  var jslibFilter = $.filter('**/' + config.optimized.lib, {restore: true});
+  var indexFilter = $.filter('**/index*.html', {restore: true});
 
   var templateCache = config.temp + config.templateCache.file;
 
-  return gulp
+  var combined = gulp
     .src(config.index)
     .pipe($.plumber())
     .pipe(inject(templateCache, 'templates'))
-    .pipe(assets) // Gather all assets from the html with useref
+
+    // Apply the concat and file replacement with useref
+    .pipe($.useref({searchPath: './'}))
 
     // Get the css
     .pipe(cssFilter)
-    .pipe($.minifyCss({processImportFrom:['!fonts.googleapis.com']}))
-    .pipe(cssFilter.restore())
+    .pipe($.sourcemaps.init())
+    .pipe($.cssnano({safe: true}))
+    .pipe($.sourcemaps.write('.'))
+    .pipe(cssFilter.restore)
 
     // Get the custom javascript
     .pipe(jsAppFilter)
+    .pipe($.sourcemaps.init())
     .pipe($.ngAnnotate({add: true}))
     .pipe($.uglify())
-    .pipe(jsAppFilter.restore())
+    .pipe($.sourcemaps.write('.'))
+    .pipe(jsAppFilter.restore)
 
     // Get the vendor javascript
     .pipe(jslibFilter)
+    .pipe($.sourcemaps.init())
     .pipe($.uglify()) // another option is to override wiredep to use min files
-    .pipe(jslibFilter.restore())
+    .pipe($.sourcemaps.write('.'))
+    .pipe(jslibFilter.restore)
 
     // Take inventory of the file names for future rev numbers
     .pipe($.rev())
-    // Apply the concat and file replacement with useref
-    .pipe(assets.restore())
-    .pipe($.useref())
     // Replace the file names in the html with rev numbers
     .pipe($.revReplace())
-    .pipe(gulp.dest(config.build));
+    // Undo rename of index.html
+    .pipe(indexFilter)
+    .pipe($.rename('index.html'))
+    .pipe(indexFilter.restore)
```

Comment:
Little hoop to make sure revReplace doesn't rename index.html to index-suffix.html, as express server contains hard-coded refs to index.html..